### PR TITLE
[examples] Adds hot potato example

### PR DESCRIPTION
**Pull request by damirka**

### CODE REVIEW

File: doc/book/src/patterns/hot-potato.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Hot Potato
 
+Hot Potato is a name for a struct that has no abilities, hence it can only packed and unpacked in its module. The pattern itself creates a case where function B has to be called after function A if A returns a potato and B consumes it.
```
Comment (randall-Mysten) [inline doc/book/src/patterns/hot-potato.md:3]:
```suggestion
Hot Potato is a name for a struct that has no abilities, hence it can only be packed and unpacked in its module. In this struct, you must call function B after function A in the case where function A returns a potato and function B consumes it.
```
The second sentence is a bit hard to parse, hopefully my proposed change is still accurate.

**Review by randall-Mysten — APPROVED** [review]

File: doc/book/examples/sources/patterns/hot-potato.move [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+    /// For when paid amount does not match the price
+    const EIncorrectAmount: u64 = 2;
+
+    /// A phone; can be purchased or traded in for a newer model
+    struct Phone has key, store { info: Info, model: u8 }
+
+    /// Payable receipt. Has to be paid directly or paid with a trade-in option.
+    /// Cannot be stored, owned or dropped - has to be used to select one of the
+    /// options for payment: `trade_in` or `pay_full`.
+    struct Receipt { price: u64 }
```
Comment (sblackshear) [inline doc/book/examples/sources/patterns/hot-potato.move:29]: Should this remember the `ID` of the `Phone` it was used to purchase?

In this particular code, it's probably harmless to mix/match phones and receipts, but I can imagine a lot of cases where you would not want to allow this.

**Comment by Clay-Mysten** [comment]
Adding @lxfind so he sees the examples growing.

File: doc/book/examples/sources/patterns/hot-potato.move (hunk 2 above)
Comment (damirka) [reply doc/book/examples/sources/patterns/hot-potato.move:29]: Didn't figure out a way to check for ID as phone is never passed afterwards. Merging this in to have this example in the book.
